plotly/tools.py

```diff
@@ -51,6 +50,15 @@ def warning_on_one_line(message, category, filename, lineno,
 except ImportError:
     _numpy_imported = False
 
+try:
+    import scipy as scp
+    import scipy.spatial as scs 
+    import scipy.cluster.hierarchy as sch 
+
+    _scipy_imported = True
+except ImportError:
+    _scipy_imported = False
```

Comment:
Yet another, other option would be to *not* automatically import this, so that user's will naturally get a Python `ImportError` when they attempt to use it because *they* need to manually import this... I think this isn't as user-friendly, but perhaps more robust.

I still think we should be *nice* to users and continue with our protected imports pattern...